### PR TITLE
fetching customer by cristin id

### PR DESCRIPTION
**Pull request by truhacevkir**
Adding getCustomerByCristinId method to IdentityServiceClient. Will use this and existing getUser methods in log generation.

### CODE REVIEW

File: clients/src/main/java/no/unit/nva/clients/IdentityServiceClient.java [hunk 1]
```diff
@@ -82,6 +85,25 @@ public GetUserResponse getUser(String userName) throws NotFoundException {
...
+                   .orElseThrow(this::handleFailure);
+    }
+
+    private URI constructCustomerGetPath(URI topLevelOrgCristinId) {
+        var customerByCristinIdUri = UriWrapper.fromHost(API_HOST)
+                                    .addChild(CUSTOMER_PATH_PARAM)
+                                    .addChild(CRISTIN_ID_PATH_PARAM)
+                                    .getUri();
+        return URI.create(
+            customerByCristinIdUri + "/" + URLEncoder.encode(topLevelOrgCristinId.toString(), UTF_8));
```
Comment (truhacevkir) [inline clients/src/main/java/no/unit/nva/clients/IdentityServiceClient.java:104]: UriWrapper does not support encoded uri in path.

File: clients/src/test/java/no/unit/nva/clients/IdentityServiceClientTest.java [hunk 2]
```diff
@@ -173,6 +174,29 @@ void shouldThrowNotFoundWhenHttpClientNotFound() throws IOException, Interrupted
         assertThrows(NotFoundException.class, action);
     }
 
+    @Test
+    void shouldReturnCustomerWhenRequested() throws NotFoundException, IOException, InterruptedException {
+        var customerCristinId = randomUri();
+        var expectedCustomer = createCustomer(customerCristinId);
+        var mockedResponse = mockResponse(expectedCustomer.toJsonString());
+        when(httpClient.send(any(HttpRequest.class), any(BodyHandler.class))).thenReturn(mockedResponse);
```
Comment (monaullah) [inline clients/src/test/java/no/unit/nva/clients/IdentityServiceClientTest.java:182]: Match expected uri instead of `any()`?

Comment (monaullah) [reply clients/src/test/java/no/unit/nva/clients/IdentityServiceClientTest.java:182]: This will ensure that the cristin id is encoded correctly

**Review by brinxmat — COMMENTED** [review]
See other reviewer's comments

**Review by brinxmat — APPROVED** [review]

File: clients/src/test/java/no/unit/nva/clients/IdentityServiceClientTest.java [hunk 3]
```diff
@@ -178,8 +184,14 @@ void shouldThrowNotFoundWhenHttpClientNotFound() throws IOException, Interrupted
     void shouldReturnCustomerWhenRequested() throws NotFoundException, IOException, InterruptedException {
         var customerCristinId = randomUri();
         var expectedCustomer = createCustomer(customerCristinId);
-        var mockedResponse = mockResponse(expectedCustomer.toJsonString());
-        when(httpClient.send(any(HttpRequest.class), any(BodyHandler.class))).thenReturn(mockedResponse);
+        var request = HttpRequest.newBuilder()
+                          .GET()
+                          .uri(createFetchCustomerByCristinIdUri(customerCristinId))
```
[the remark this answers is not in the document]
Comment (truhacevkir) [reply clients/src/test/java/no/unit/nva/clients/IdentityServiceClientTest.java:189]: Not as I can see from openapi scheme.

**Review by monaullah — APPROVED** [review]

**Pull request merged**